sentinel/lib/messages.js

```diff
@@ -1,228 +1,115 @@
-        } else if (doc.fields && doc.fields[recipient]) {
-            // Try to resolve a specified property/field name
-            phone = doc.fields[recipient];
-        } else if (doc[recipient]) {
-            // Or directly on the doc
-            phone = doc[recipient];
-        } else if (recipient.indexOf('.') > -1) {
-            // Or multiple layers by executing it as a statement
-            phone = objectPath.get(doc, recipient);
+        // use the translation key if provided
+        if (configuration.translation_key) {
+            return utils.translate(configuration.translation_key, locale);
         }
 
-        return phone || _default || doc.from;
-    },
-    addMessage: addMessage,
-    getMessage: getMessage,
-    notifyGrandparent: function(doc, message, options) {
-        var self = module.exports;
-        addMessage({
-            doc: doc,
-            message: message,
-            phone: self.getRecipientPhone(doc, 'grandparent_phone'),
-            templateContext: options
-        });
-    },
-    notifyParent: function(doc, message, options) {
-        var self = module.exports;
-        addMessage({
-            doc: doc,
-            message: message,
-            phone: self.getRecipientPhone(doc, 'parent_phone'),
-            templateContext: options
-        });
+        // otherwise, use the configured messages (deprecated)
+        var messages = configuration.messages || configuration.message;
+        if (!_.isArray(messages)) {
+            console.warn('Message property should be an array. Please check your configuration.');
+            return '';
+        }
+        if (!messages.length) {
+            console.warn('Message property array was empty. Please check your configuration.');
+            return '';
+        }
+        // default to first item in messages array in case locale match fails
+        var message = _.findWhere(messages, { locale: locale || 'en' }) || messages[0];
+        return (message.content && message.content.trim()) || '';
     },
-    addReply: function(doc, message, options) {
-        var self = module.exports;
-        addMessage({
-            doc: doc,
-            message: message,
-            phone: self.getRecipientPhone(doc, 'clinic'),
-            templateContext: options
-        });
+    /*
+     * Return false when the recipient phone matches the denied list.
+     *
+     * outgoing_deny_list is a comma separated list of strings. If a string in
+     * that list matches the beginning of the phone then we set up a response
+     * with a denied state. The pending message process will ignore these
+     * messages and those reports will be left without an auto-reply. The
+     * denied messages still show up in the messages export.
+     *
+     * @param {String} from - Recipient phone number
+     * @returns {Boolean}
+     */
+    isOutgoingAllowed: from => {
+      const conf = config.get('outgoing_deny_list') || '';
```

Comment:
This could be shifted to below the first two short-cuttings.

Reply:
Also instead of defining a missing config as `''`, couldn't we just return `true`?

Reply:
Done.